### PR TITLE
Update DG with details done by Wu Qirui

### PR DESCRIPTION
**Pull request by hhdqirui**
Added parts about `editLocation` and `deleteLocation`.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -533,7 +545,10 @@ This feature is one of the different `list` commands. It has the same command wo
 * Infected person is defined as person with infection status as `true`
 * Infected visit is defined as visit made by any infected person
 * Infected location is defined as location that has been visited by any infected person
-* Number of high risk location of infection is defined as:    
+* If user specifies number of high risk locations in the command, then number of high risk locations would be the user 
+specified value.
+* If user does not specifies number of high risk locations, then number of high risk location of infection is 
+determined using the following rule:    
     * If number of infected location is larger than 60% of number of total number of location, then the number of high risk
```
Comment (siangernlow) [inline docs/DeveloperGuide.md:552]: This part is quite technical, so you could perhaps consider adding some diagrams? A simple diagram would suffice. Alternatively, you may provide an example with concrete numbers to bring the idea across more clearly.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -562,19 +582,135 @@ The following activity diagram summarizes what happens when a user executes the
...
-##### Aspect: Definition of number of high risk location of infection
-When displaying the list of high risk location of infection, the top few locations that has been most visited by any 
-infected people are wanted. 
+##### Aspect: User input for number of high risk locations
+The parameter for list high risk locations command is optional. Users can choose to or choose not to input the number of
+high risk locations for this command. 
+
+**Rationale**
+Instead of always using the pre-defined rule within VirusTracker, this implementation allows users to customize the
+number of high risk locations displayed in VirusTracker. If the pre-defined rule is in used, users might not be able to
```
Comment (siangernlow) [inline docs/DeveloperGuide.md:591]: Small grammatical error. `... is in used` -> `is in use`.

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -562,19 +582,135 @@ The following activity diagram summarizes what happens when a user executes the
...
+The parameter for list high risk locations command is optional. Users can choose to or choose not to input the number of
+high risk locations for this command. 
+
+**Rationale**
+Instead of always using the pre-defined rule within VirusTracker, this implementation allows users to customize the
+number of high risk locations displayed in VirusTracker. If the pre-defined rule is in used, users might not be able to 
+view more infected locations beyond the displayed high risk locations that are selected automatically by VirusTracker.
+
+##### Aspect: Determining number of high risk locations for infection when user does not specify the number
+In the case when user does not specify the number of high risk locations in the command, the system itself will
```
Comment (siangernlow) [inline docs/DeveloperGuide.md:595]: Small grammatical error. `In the case when the user...`

File: docs/diagrams/DeleteVisitActivityDiagram.puml [hunk 4]
```diff
@@ -4,7 +4,7 @@ start
 'Since the beta syntax does not support placing the condition outside the
 'diamond we place it as the true branch instead.
 
-repeat :User executes deleteVisit command;
+repeat :User executes deleteLocation command;
```
Comment (siangernlow) [inline docs/diagrams/DeleteVisitActivityDiagram.puml:7]: Is this supposed to be deleteVisit instead? It is in the DeleteVisitActivityDiagram.puml file.

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (hhdqirui) [reply docs/DeveloperGuide.md:552]: Examples added.

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (hhdqirui) [reply docs/DeveloperGuide.md:591]: Fixed.

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (hhdqirui) [reply docs/DeveloperGuide.md:595]: Fixed.

File: docs/diagrams/DeleteVisitActivityDiagram.puml (hunk 4 above)
Comment (hhdqirui) [reply docs/diagrams/DeleteVisitActivityDiagram.puml:7]: It is a mistake. Fixed.

**Review by siangernlow — APPROVED** [review]
LGTM

**Pull request merged**